src/plugins/intel_gpu/src/kernel_selector/core/actual_kernels/eddo/experimental_detectron_detection_output_kernel_ref.cpp

```diff
@@ -0,0 +1,210 @@
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "experimental_detectron_detection_output_kernel_ref.h"
+
+#include <algorithm>
+#include <string>
+
+#include "kernel_selector_utils.h"
+
+namespace kernel_selector {
+
+ParamsKey ExperimentalDetectronDetectionOutputKernelRef::GetSupportedKey() const {
+    ParamsKey k;
+    k.EnableInputDataType(Datatype::F16);
+    k.EnableInputDataType(Datatype::F32);
+    k.EnableInputDataType(Datatype::INT32);
+    k.EnableInputDataType(Datatype::INT64);
+    k.EnableOutputDataType(Datatype::F16);
+    k.EnableOutputDataType(Datatype::F32);
+    k.EnableOutputDataType(Datatype::INT32);
+    k.EnableOutputDataType(Datatype::INT64);
+    k.EnableInputLayout(DataLayout::bfyx);
+    k.EnableOutputLayout(DataLayout::bfyx);
+    k.EnableBatching();
+    k.EnableDifferentTypes();
+    return k;
+}
+
+KernelsPriority ExperimentalDetectronDetectionOutputKernelRef::GetKernelsPriority(const Params&,
+                                                                                  const optional_params&) const {
+    return DONT_USE_IF_HAVE_SOMETHING_ELSE;
+}
+
+bool ExperimentalDetectronDetectionOutputKernelRef::Validate(const Params& p, const optional_params& o) const {
+    if (p.GetType() != KernelType::EXPERIMENTAL_DETECTRON_DETECTION_OUTPUT ||
+        o.GetType() != KernelType::EXPERIMENTAL_DETECTRON_DETECTION_OUTPUT) {
+        return false;
+    }
+    return true;
+}
+
+constexpr int kBoxesInputIdx = 0;
+constexpr int kDeltasInputIdx = 1;
+constexpr int kScoresInputIdx = 2;
+constexpr int kImInfoInputIdx = 3;
+constexpr int kOutputClassesInputIdx = 4;
+constexpr int kOutputScoresInputIdx = 5;
+
+constexpr int kRefinedBoxesBufferIdx = 0;
+constexpr int kRefinedBoxAreasBufferIdx = 1;
+constexpr int kRefinedScoresBufferIdx = 2;
+constexpr int kScoreClassIndexBufferIdx = 3;
+constexpr int kDetectionCountBufferIdx = 4;
+
+constexpr int kBufferCount = 5;
+
+constexpr int kOutputIdx = 0;
+
+constexpr int kRoiCountScalarIdx = 0;
+
+JitConstants ExperimentalDetectronDetectionOutputKernelRef::GetJitConstants(
+    const experimental_detectron_detection_output_params& params) const {
+    JitConstants jit = MakeBaseParamsJitConstants(params);
+
+    jit.AddConstants({
+        MakeJitConstant("SCORE_THRESHOLD", params.score_threshold),
+        MakeJitConstant("NMS_THRESHOLD", params.nms_threshold),
+        MakeJitConstant("NUM_CLASSES", params.num_classes),
+        MakeJitConstant("POST_NMS_COUNT", params.post_nms_count),
+        MakeJitConstant("MAX_DETECTIONS_PER_IMAGE", params.max_detections_per_image),
+        MakeJitConstant("MAX_DELTA_LOG_WH", params.max_delta_log_wh),
+        MakeJitConstant("DELTA_WEIGHT_X", params.deltas_weights[0]),
+        MakeJitConstant("DELTA_WEIGHT_Y", params.deltas_weights[1]),
+        MakeJitConstant("DELTA_WEIGHT_LOG_W", params.deltas_weights[2]),
+        MakeJitConstant("DELTA_WEIGHT_LOG_H", params.deltas_weights[3]),
+
+        MakeJitConstant("OUTPUT_TYPE1", "int"),
```

Comment:
maybe better to have less general  const name to avoid conflicts in future? what about int64? (maybe better to have sth like
"OUTPUT_INDICES_TYPE", "INPUT7_TYPE"  (corresponded input index)

Reply:
Good point, done. I believe though you meant INPUT5_TYPE, not INPUT7_TYPE

Reply:
Actually, it's even INPUT4_TYPE